Userland/Applications/CertificateSettings/CertificateStore.cpp

```diff
@@ -59,6 +68,43 @@ GUI::Variant CertificateStoreModel::data(GUI::ModelIndex const& index, GUI::Mode
     return {};
 }
 
+ErrorOr<size_t> CertificateStoreModel::try_add(Vector<Certificate> const& certificates)
+{
+    auto size = m_certificates.size();
+    TRY(m_certificates.try_extend(certificates));
+    return m_certificates.size() - size;
+}
+
+ErrorOr<void> CertificateStoreWidget::try_import()
+{
+    auto file = FileSystemAccessClient::Client::the().open_file(window(), "Choose PEM to import...");
+    if (file.is_error())
+        return {};
+
+    auto filename = file.value().filename();
+    if (!(filename.ends_with_bytes(".pem"sv) || filename.ends_with_bytes(".crt"sv)))
+        return Error::from_string_view("File is not a .pem or .crt file."sv);
+
+    auto data = TRY(file.release_value().release_stream()->read_until_eof());
+    auto count = TRY(m_root_ca_model->try_add(TRY(DefaultRootCACertificates::the().reload_certificates(data))));
```

Comment:
Just to confirm, `reload_certificates` as written will not clear out the old certs until the new ones have been properly parsed right? Would be bad to have something like this brick the certs of the system:

```
$ echo "lmao" > totally_a_cert.pem
```

Reply:
`reload_certificates` will just return you a `Vector<Certificate>`, it's on you to decide what to do.

Reply:
Also, the `cacert.pem` containing the systems CAs will not be written to at all.